ogr/abstract.py

```diff
@@ -281,6 +280,7 @@ def get_project(self, **kwargs) -> "GitProject":
 
     def get_project_from_url(self, url: str) -> "GitProject":
         repo_url = parse_git_repo(potential_url=url)
+        assert repo_url, f"Failed to return repo for url: {url}"
```

Comment:
The assert should not be in the real code -- it is meant for testing and development. (The code can be run in such way, that the `assert` statements are skipped.)

Reply:
Do you want me to remove this line? I put it in for type hinting reasons because output from `parse_git_repo` can be `None` (unless we assert that it isn't) and so mypy complains that we're trying to access the attributes of a NoneType in the line below this. Without the assertion I think you would get an AttributeError if the parsing returned a None, which seems to be a valid path. 

Or did you mean that a different error type should be thrown here in the None case so that the assertion cannot be skipped?

Reply:
Thanks for creating the issue -- let's leave it as is (with `Any`).

Reply:
Ideally, we want to fix that -- test the `repo_url` to `not None` and raise an `OgrException` otherwise.

(So we can remove the `assert`.)